File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution { //shraddha
public: 
    int maxProfit(vector<int>& prices) {
        int maxprofit = 0;
        int bestbuy = prices[0];
        for( int i =1 ; i< prices.size(); i++ ){
             if(prices[i] > bestbuy){
                maxprofit = max(prices[i] - bestbuy,maxprofit);
             }

            bestbuy = min(bestbuy,prices[i]);
        }
        return maxprofit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
public int maxProfit(int[] prices) {
   
     int buyP = prices[0];
     int profit = 0;

    for(int i = 1; i< prices.length; i++){
        if(buyP > prices[i]){
            buyP = prices[i];
        }
        else{
            int currp = prices[i] - buyP;
            profit = Math.max(profit,currp); 
        }
    }

    return profit;
    }
}